generator/RecordGen.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Xml;
+
+namespace GtkSharp.Generation
+{
+	public class RecordGen : GenBase
+	{
+		// Logic around Record is that it's a struct generation
+		// If it has no fields it is an Opaque gen
```

Comment:
Probably, but usually these structs are already marked as disguised or private so I don't think additional logic is needed on the generator side. If there's a public struct with no field then it's probably a bug on the native library / giscanner side.

Reply:
Also I think OpaqueGen should be rewritten as it relies heavily on glue code right now for field access which could be replaced by marshalling if the struct layout is available.

Reply:
Is this it? https://github.com/shana/bindinator/blob/master/scheme/gir.xsd If so, we can use this and make the deserialization way easier.

Reply:
That one is outdated. I'll upload a more recent one to my gi-sharp repository

Reply:
https://github.com/stsundermann/gi-sharp/commit/9ba2c750427311220d238e5d39d1dfa2997901d6